src/exec_auth.ts

```diff
@@ -99,15 +99,33 @@ export class ExecAuth implements Authenticator {
             exec.env.forEach((elt) => (env[elt.name] = elt.value));
             opts = { ...opts, env };
         }
-        const result = this.execFn(exec.command, exec.args, opts);
-        if (result.error) {
-            throw result.error;
-        }
-        if (result.status === 0) {
-            const obj = JSON.parse(result.stdout.toString('utf8')) as Credential;
-            this.tokenCache[user.name] = obj;
-            return obj;
-        }
-        throw new Error(result.stderr.toString('utf8'));
+
+        return new Promise((accept, reject) => {
+            let stdoutData: string = '';
+            let stderrData: string = '';
+
+            const subprocess = this.execFn(exec.command, exec.args, opts);
+
+            subprocess.stdout.on('data', (data: Buffer | string) => {
+                stdoutData += data.toString('utf8');
```

Comment:
I think the `data.toString('utf8')` line here and on line 114 may be subtly incorrect.

With this logic, chunks are emitted as `Buffer`s, which are then converted to utf8 strings. However, utf8 characters can be multiple bytes, and the `Buffer` based logic does not know anything about that. It's possible that a utf8 character gets split across multiple chunks, which results in the wrong final string.

I would suggest updating this code in one of two ways:
- Do all of the chunk collecting with `Buffer`s and then convert everything to utf8 at once in the `'close'` event.
- Use `subprocess.std{out,err}.setEncoding('utf8')` on line 108 so that Node emits `'data'` events with proper utf8 chunks.

Reply:
ok, thanks, I'll change with the second option